new/collapse-mixlib-install-in-tk.md

```diff
@@ -0,0 +1,171 @@
+
+##### New Settings
+* `product_version`: Manage `latest` and `<version>` values.  Supports partial versions.
+* `install_strategy`: `once`, `skip`, `always`. Default: `once`
+  * `once`: Behaves similar to true. Install `product_name` `product_version` once. Don't install `product_name` if detected.
+  * `skip`: Behaves similar to false. Do not install `product_name`.
+  * `always`: Behaves similar to latest. Always install specified/latest `product_name` `product_version`.
+
+###### Examples
+```
+require_chef_omnibus: true
+```
+equivalent to:
+```
+product_version: latest
+install_strategy: once
+```
+_Default settings_
+
+----
+
+```
+require_chef_omnibus: false
+```
+equivalent to:
+```
+install_strategy: skip
+```
+
+----
+
+```
+require_chef_omnibus: latest
+```
+equivalent to:
+```
+product_version: latest
+install_strategy: always
+```
+
+----
+
+```
+require_chef_omnibus: 12
+```
+equivalent to:
+```
+product_version: 12
+install_strategy: once
+```
+
+###### Additional Flexibility
+```
+product_version: 12
+install_strategy: always
+```
+
+##### Future
+We could add an upgrade install strategy which only installs the package if a newer package is available.
+
+###### Examples
+```
+product_version: latest
+install_strategy: upgrade
+```
+
+----
+
+```
+product_version: 12.19
+install_strategy: upgrade
+```
+
+#### Opportunity for improvement
+Although adding install_strategy will provide feature parity, we should consider what features we really need in Kitchen moving forward. We may not need this option if we come up with the use cases we feel Kitchen should support moving forward.
+Does Kitchen need to support the same require_chef_omnibus configurabilty with the new API?
+
+By simplifying the use cases, can we automate the install and upgrade paths without introducing a new setting?
+
+#### Mixlib Install changes
+A new script param, `install_strategy`, will be added to `install.sh` and `install.ps1`. The option will determine how the install functions will install the package. The script params will default to always which is the current behavior.
+
+Adding `install_strategy` to the install scripts moves all responsibility for determining how to install packages to a canonical source. This also decouples the logic in Kitchen for determing installs and upgrades from Mixlib Install's ScriptGenerator.
+
+----
+
+Other provisioner option examples like `chef_omnibus_install_options`, `chef_omnibus_url`, `chef_metadata_url`, `chef_omnibus_root`, and `ruby_bindir` have been shown how they can be deprecated in this __experimental__ branch: https://github.com/test-kitchen/test-kitchen/compare/master...wrightp:pw/config-deprecations. Most of them are simple.
+
+To further separate new from old functionality, none of the options should cross over to one another. We will discuss any exceptions as they arise. `product_name` will act as the pivot point between which code path to follow (`ScriptGenerator` or the new API).
```

Comment:
Does "crossing over" mean that distinct option names and code paths will be maintained for the old functionality and the new?